lib/puppet_x/bodeco/util.rb

```diff
@@ -87,6 +88,10 @@ def generate_request(uri)
         header = @cookie && { 'Cookie' => @cookie }
 
         request = Net::HTTP::Get.new(uri.request_uri, header)
+        @headers.each do |h|
+          h_split = h.split(':')
+          request[h_split[0].strip] = h_split[1].strip if h_split.length == 2
```

Comment:
This ignore headers with a `:` in their value:

```suggestion
          h_split = h.split(':', 2)
```

But maybe it makes more sense to directly use a Hash for `@headers`?  Or are there limitations i am not thinking about?

Reply:
There is already headers parameter for the archive type: https://github.com/voxpupuli/puppet-archive/blob/master/lib/puppet/type/archive.rb#L207